### ISSUE TITLE
Custom Permissions on Custom Plugins not set to true by default for Super Admins

### ISSUE THREAD

**Issue by 100KILOFUNK** [strapi/strapi#17754]
### Required System information

- Node.js version: 18.5.0
- NPM version: 9.5.0
- Strapi version: 4.11.5
- Database: postgres
- Operating system: Ubuntu
- Is your project Javascript or Typescript: JS

### Describe the bug

I've added a custom RBAC action to the bootstrap lifecycle, registered it and tied it to a custom plugin. My intention was that Super Admin and Editor roles should have access to the plugin, while authors do not. The issue however is that the action is not ticked for the super admin by default, and cannot be ticked since it is greyed out.

### Steps to reproduce the behavior

1. Register custom RBAC action in the bootstrap lifecycle function in `src/index.js`

```js
// Role Based Access Control
const actions = [
  {
    uid: "read",
    displayName: "Read",
    pluginName: "site-publisher",
    section: "plugins",
  },
];

...

async bootstrap() {
  await strapi.admin.services.permission.actionProvider.registerMany(actions);
```

2. Add the permission to the custom plugin: `backend/src/plugins/name-of-plugin/admin/src/index.js`

```js

        return component;
      },
      permissions: [
        {
          action: 'plugin::site-publisher.read', // the action name should be plugin::plugin-name.actionType
          subject: null,
        },
      ],
    });
    app.registerPlugin({
      id: pluginId,
```

3. Rebuild and start
4. Permission is not ticked. See image under screenshots.

### Expected behavior

I expect the box to be ticked!

### Screenshots

![image](https://github.com/strapi/strapi/assets/13716369/c3390404-7812-4f15-b7c3-7b192ddc4f7b)

### Additional context

Permission is working and being set for other roles fine.

**Comment by 100KILOFUNK** [comment]
I rebuilt the solution to be more in line with best practice, and wrote an article about it (sorry about the documentation complaint in the intro): https://medium.com/@n.goniband/rbac-permissions-for-custom-plugins-strapi-4-4680232052f4

The issue persists however, can this have to do with running strapi in a serverless environment? In my black-box-mind, this has to do with database changes not being persistent.

Any updates? This is critical for our solution.

**Comment by viniciusFelipeS** [comment]
I went through the same problem and solved it by adding the bootstrap related to the plugin directly in the plugin.

Instead of making the bootstrap call inside `src/index.js`,
I did it inside `src\extensions\plugin-name\strapi-server.js`

like this:

```javascript
"use strict";

module.exports = async (plugin) => {
  plugin.bootstrap = async ({ strapi }) => {
    const actions = [
      {
        section: "plugins",
        displayName: "Read",
        uid: "read",
        pluginName: "plugin-name",
      },
    ];

    await strapi.admin.services.permission.actionProvider.registerMany(actions);
  };

  return plugin;
};
```

and it worked perfectly

**Comment by H-Gomez** [comment]
I am experiencing this issue with Strapi V4 even after performing the steps above, as well as the steps found in various places on the forum. The custom plugin permissions shows under **Settings > Roles > Super Admin > Plugins** in the admin dashboard. It has `READ `checked.

However, when I deploy to production this is **unchecked**, and all boxes are disabled for super admin (read-only in production as per Strapi docs). Do you know why it works in Localhost but not in production? 

Localhost:
![image](https://github.com/user-attachments/assets/7008440c-890a-4605-a3cf-31f0857cbea0)

Production
![image](https://github.com/user-attachments/assets/b7dfd925-f98b-4ded-9a7c-b20ff21667e6)

**Comment by mallowsc** [comment]
This may sound silly but did you rebuild after moving it to production?

**Comment by H-Gomez** [comment]
> This may sound silly but did you rebuild after moving it to production?

Yep. Our build pipeline runs the `build` command for each plugin and also the root Strapi as well as installs the node modules for each plugin.
